lib/groonga-query-log/command/run-regression-test.rb

```diff
@@ -571,6 +644,75 @@ def use_persistent_cache?
           @old.use_persistent_cache? or @new.use_persistent_cache?
         end
       end
+
+      class EMailNotifier
+
+        def initialize(tester_result, start_time, notifier_options)
+          @start_time = start_time
+          @tester_result = tester_result
+          @options = notifier_options
+        end
+
+        def elapsed_time_format
+          elapsed = Time.now - @start_time
+          elapsed_seconds = elapsed % 60
+          elapsed_minutes = elapsed / 60 % 60
+          elapsed_hours = elapsed / 60 / 60 % 24
+          elapsed_days = elapsed / 60 / 60 / 24
+          sprintf("Elapsed: %ddays %02d:%02d:%02d\n",
+                 elapsed_days,
+                 elapsed_hours,
+                 elapsed_minutes,
+                 elapsed_seconds)
+        end
+
+        def run
+          $stdout = Tempfile.open("format_stdout")
+          command = GroongaQueryLog::Command::FormatRegressionTestLogs.new
```

Comment:
We should not swap `$stdout`. We should extract a class from `FormatRegressionTestLogs` and the extracted class should accept the output.

Reply:
It should be done in another pull request before this pull request.